lib/FrontendTool/ImportedModules.cpp

```diff
@@ -66,9 +66,19 @@ bool swift::emitImportedModules(ModuleDecl *mainModule,
     if (!ID)
       continue;
 
-    auto modulePath = ID->getModulePath();
-    // only the top-level name is needed (i.e. A in A.B.C)
-    Modules.insert(modulePath[0].Item.str());
+    // Check if module aliasing was used for an imported module; for example,
+    // if '-module-alias Foo=Bar' was passed and this module has 'import Foo',
+    // its corresponding real module name 'Bar' should be emitted.
+    ImportPath::Module::Builder builderWithRealModuleName;
+    auto modulePath = ID->getModulePath(/*outRealModuleName=*/&builderWithRealModuleName);
+
+    if (!builderWithRealModuleName.get().empty()) {
+      // Module aliasing was used, so use its real module name
+     Modules.insert(builderWithRealModuleName.get().front().Item.str());
+    } else {
+      // only the top-level name is needed (i.e. A in A.B.C)
+      Modules.insert(modulePath[0].Item.str());
+    }
```

Comment:
Again, this becomes much shorter with my suggestion:

```suggestion
    ImportPath::Module::Builder scratch;
    auto modulePath = ID->getRealModulePath(scratch);
    // only the top-level name is needed (i.e. A in A.B.C)
    Modules.insert(modulePath[0].Item.str());
```